### PR TITLE
Feat/do array exercises

### PR DESCRIPTION
**Pull request by CBinh710**
## Do array exercises
issue: [link](https://github.com/users/CBinh710/projects/8/views/1?pane=issue&itemId=80165780)

image
image

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,68 @@
-function exampleFunction(arg1, arg2) {
-  const result = arg1 + arg2;
-  console.log('Result:', result);
-  return result;
+// Exercise 1
+var randomValues = [1, '1', 'Mike', 0, 12, true];
```
Comment (phuocduong-agilityio) [inline src/index.js]: var?

File: src/index.js [hunk 2]
```diff
@@ -1,6 +1,68 @@
...
+
+// 1. Print type of all items in randomValues
+randomValues.forEach((value) => console.log(typeof value));
+
+// 2. How many number in randomValues
+const numberCount = randomValues.filter((valueNumber) => typeof valueNumber === 'number').length;
```
Comment (phuocduong-agilityio) [inline src/index.js:9]: Is there other solution?

File: src/index.js [hunk 3]
```diff
@@ -1,6 +1,68 @@
...
+console.log(numberCount);
+
+// Exercise 2
+var randomNumbers = [10, 20, 50, 25, 70, 0, 55, 90];
+// 1. Find the item with value = 70 and 30 in randomNumbers. Find at least 4 ways to do this
+
+// Using indexOf()
+let index30 = randomNumbers.indexOf(30);
```
Comment (phuocduong-agilityio) [inline src/index.js]: Let?

File: src/index.js [hunk 4]
```diff
@@ -1,6 +1,68 @@
...
+let index70 = randomNumbers.indexOf(70);
+console.log(index30);//-1
```
Comment (phuocduong-agilityio) [inline src/index.js]: Could you provide more infor to log?

File: src/index.js [hunk 5]
```diff
@@ -1,6 +1,68 @@
...
+console.log(index70);//5
+
+// Using find()
+let item30 = randomNumbers.find(value => value === 30);
+let item70 = randomNumbers.find(value => value === 70);
+console.log(item30);
```
Comment (phuocduong-agilityio) [inline src/index.js:38]: Sometimes, you write result. Sometimes not

File: src/index.js [hunk 6]
```diff
@@ -1,6 +1,68 @@
...
+console.log(item70);
+
+// Using filter()
+// let index30 = randomNumbers.filter(value => value === 30);
```
Comment (phuocduong-agilityio) [inline src/index.js]: Why comment it?

File: src/index.js [hunk 7]
```diff
@@ -1,6 +1,68 @@
...
+console.log(index30);//-1
+console.log(index70);//5
+
+// Using find()
+let item30 = randomNumbers.find(value => value === 30);
+let item70 = randomNumbers.find(value => value === 70);
+console.log(item30);
+console.log(item70);
+
+// Using filter()
```
Comment (phuocduong-agilityio) [inline src/index.js:41]: What is its purpose to use filter?

File: src/index.js [hunk 8]
```diff
@@ -1,6 +1,68 @@
...
+// 1. Find the item with value = 70 and 30 in randomNumbers. Find at least 4 ways to do this
+
+// Using indexOf()
+let index30 = randomNumbers.indexOf(30);
+let index70 = randomNumbers.indexOf(70);
+console.log(index30);//-1
+console.log(index70);//5
+
+// Using find()
+let item30 = randomNumbers.find(value => value === 30);
```
Comment (phuocduong-agilityio) [inline src/index.js]: After let/const add empty line

File: src/index.js [hunk 9]
```diff
@@ -1,6 +1,68 @@
...
+}
+console.log(maxNumber);
+console.log(maxIndex);
+
+// 3. Plus each number in randomNumbers with 5. Result should be [15, 25, 55 …]. Find at least 2 ways to do this
+// Using Map()
+let plusFive = randomNumbers.map(num => num + 5);
+console.log(plusFive);
+// Using loop
+let plusFiveLoop = [];
```
Comment (phuocduong-agilityio) [inline src/index.js:84]: How about foreach?

File: src/index.js (hunk 1 above)
Comment (CBinh710) [reply src/index.js]: it should be `const` because on this exercise the array is not changed

File: src/index.js (hunk 2 above)
Comment (CBinh710) [reply src/index.js:9]: I use for loop and conditional

File: src/index.js (hunk 3 above)
Comment (CBinh710) [reply src/index.js]: I should use `const` because it used for value of a variable does not change after it is assigned.

File: src/index.js (hunk 4 above)
Comment (CBinh710) [reply src/index.js]: Logging in JavaScript is an important practice for debugging and understanding the flow of code. JavaScript provides several built-in methods through the console object that allow you to log messages, objects, warnings, errors, and more. I will detail it in Notebook

File: src/index.js (hunk 5 above)
Comment (CBinh710) [reply src/index.js:38]: I fixed all, thanks

File: src/index.js (hunk 6 above)
Comment (CBinh710) [reply src/index.js]: I had some mistakes on naming variable, i fixed that, thanks

File: src/index.js (hunk 7 above)
Comment (CBinh710) [reply src/index.js:41]: Filter used to create a new array with elements that pass a test implemented by a function. In this case i use to determine if the values ​​30 and 70 are present in the array or not. The result will be an empty array if the values ​​are not present.

File: src/index.js (hunk 8 above)
Comment (CBinh710) [reply src/index.js]: Thanks, i fixed all

File: src/index.js (hunk 9 above)
Comment (CBinh710) [reply src/index.js:84]: I can be used on this, i added it

**Pull request closed without merging**